torchtext/models/multiheadattention.py

```diff
@@ -0,0 +1,236 @@
+import torch
+from torch._jit_internal import Tuple, Optional
+
+
+Tensor = torch.Tensor
+
+
+class MultiheadAttentionContainer(torch.nn.Module):
+    def __init__(self, in_proj_tuple, attention_layer, out_proj):
+        r""" A multi-head attention container
+
+        Args:
+            in_proj_tuple: A tuple of multi-head in-projection layers
+            attention_layer: The attention layer.
+            out_proj: The multi-head out-projection layer
+
+        Examples::
+            >>> embed_dim, num_heads, bsz = 10, 5, 64
+            >>> MHA = MultiheadAttentionContainer((MultiheadInProject(embed_dim, num_heads),
+                                                   MultiheadInProject(embed_dim, num_heads),
+                                                   MultiheadInProject(embed_dim, num_heads)),
+                                                   ScaledDotProduct(),
+                                                   MultiheadOutProject(embed_dim // num_heads, num_heads))
+            >>> query = torch.rand((21, bsz, embed_dim))
+            >>> key = value = torch.rand((16, bsz, embed_dim))
+            >>> attn_output, attn_weights = MHA(query, key, value)
+            >>> print(attn_output.shape)
+            >>> torch.Size([21, 64, 10])
+        """
+        super(MultiheadAttentionContainer, self).__init__()
+        self.query_in_proj = in_proj_tuple[0]
+        self.key_in_proj = in_proj_tuple[1]
+        self.value_in_proj = in_proj_tuple[2]
```

Comment:
Can you explain what's the original idea on passing a tuple of modules here, instead of three modules or a single module?